### ISSUE TITLE
clang-sys 0.29.3 fails to build on nightly

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/rust#84162]
I tried building `clang-sys 0.29.3` with feature `runtime` on the latest nightly.

I expected to see this happen: it builds successfully.

Instead, this happened: 
```console
> cargo +nightly build --features runtime |& head -n 20
   Compiling clang-sys v0.29.3 (/tmp/tmp.EH8ss9nTlb/clang-sys-0.29.3)
error[E0642]: patterns aren't allowed in methods without bodies
    --> src/lib.rs:1660:39
     |
1660 |     pub fn clang_CXCursorSet_contains(set: CXCursorSet, cursor: CXCursor) -> c_uint;
     |                                       ^^^
     |
help: give this argument a name or use an underscore to ignore it
     |
1660 |     pub fn clang_CXCursorSet_contains(_: CXCursorSet, cursor: CXCursor) -> c_uint;
     |                                       ^

error[E0642]: patterns aren't allowed in methods without bodies
    --> src/lib.rs:1660:57
     |
1660 |     pub fn clang_CXCursorSet_contains(set: CXCursorSet, cursor: CXCursor) -> c_uint;
     |                                                         ^^^^^^
     |
help: give this argument a name or use an underscore to ignore it
     |
```

### Meta[
`rustc --version --verbose`:
```
rustc 1.53.0-nightly (d0695c908 2021-04-12)
binary: rustc
commit-hash: d0695c9081b16077d0aed368bccaf437d77ff497
commit-date: 2021-04-12
host: x86_64-unknown-linux-gnu
release: 1.53.0-nightly
LLVM version: 12.0.0
```

**Comment by Nemo157** [comment]
First regressed in `rustc 1.53.0-nightly (a86612456 2021-04-11)`.

**Comment by Aaron1011** [comment]
Will be fixed by https://github.com/rust-lang/rust/pull/84130

**Comment by SimonSapin** [comment]
This affects Servo too, through rust-bindgen:

searched nightlies: from nightly-2021-04-10 to nightly-2021-04-12
regressed nightly: nightly-2021-04-12
searched commits: from https://github.com/rust-lang/rust/commit/a836d9b6413d9d593be6c09463ff8c4c70e56599 to https://github.com/rust-lang/rust/commit/a8661245649f3d1c0dc5b23270bdac0bbd2d8f64
regressed commit: https://github.com/rust-lang/rust/commit/ba6275b6a9fc05fd5d93220e9a67fe64d663cb62

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.6.0</summary>


Host triple: x86_64-unknown-linux-gnu
Reproduce with:
```bash
cargo bisect-rustc -- check --features runtime 
```
</details>

**Comment by apiraino** [comment]
Assigning `P-critical` for its impact, as discussed as part of the [Prioritization Working Group procedure](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/.2384162.20clang-sys.200.2E29.2E3.20fails.20to.20build.20on.20nightly) and removing `I-prioritize`.

@rustbot label -I-prioritize +P-critical

**Comment by pnkfelix** [comment]
For the record, according to the bisection performed by @SimonSapin above, this was injected by PR #82608. (But as @Aaron1011 noted above, it will be fixed by PR #84130.)
